examples/cpp/rtps_entities/ReaderApp.cpp

```diff
@@ -0,0 +1,239 @@
+            eprosima::fastcdr::EncodingAlgorithmFlag::PLAIN_CDR,
+            [&data](eprosima::fastcdr::Cdr& dcdr, const eprosima::fastcdr::MemberId& mid) -> bool
+            {
+                bool ret_value = true;
+                switch (mid.id)
+                {
+                    case 0:
+                            dcdr >> data.index();
+                        break;
+
+                    case 1:
+                            dcdr >> data.message();
+                        break;
+
+                    default:
+                        ret_value = false;
+                        break;
+                }
+                return ret_value;
+            });
+}
+
+using namespace eprosima::fastdds;
+using namespace eprosima::fastdds::rtps;
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace rtps_entities {
+
+ReaderApp::ReaderApp(
+        const CLIParser::rtps_entities_config& config,
+        const std::string& topic_name)
+    : samples_(config.samples)
+    , rtps_participant_(nullptr)
+    , rtps_reader_(nullptr)
+    , reader_history_(nullptr)
+    , stop_(false)
+    , data_(new HelloWorld)
+{
+    // Create RTPS Participant
+    RTPSParticipantAttributes part_attr;
+    part_attr.builtin.discovery_config.discoveryProtocol = DiscoveryProtocol::SIMPLE;
+    part_attr.builtin.use_WriterLivelinessProtocol = true;
+    rtps_participant_ = RTPSDomain::createParticipant(0, part_attr);
+
+    if (rtps_participant_ == nullptr)
+    {
+        throw std::runtime_error("RTPS Participant creation failed");
+    }
+
+    // Reader History Attributes
+    HistoryAttributes hatt;
+    hatt.payloadMaxSize = 255;
+    reader_history_ = new ReaderHistory(hatt);
+
+    // Create RTPS Reader
+    ReaderAttributes reader_att;
+    reader_att.endpoint.reliabilityKind = RELIABLE;
+    reader_att.endpoint.durabilityKind = TRANSIENT_LOCAL;
+
+    rtps_reader_ = RTPSDomain::createRTPSReader(rtps_participant_, reader_att, reader_history_, this);
+
+    if (rtps_reader_ == nullptr)
+    {
+        throw std::runtime_error("RTPS Reader creation failed");
+    }
+
+    if (!register_entity(topic_name))
+    {
+        throw std::runtime_error("Entity registration failed");
+    }
+}
+
+ReaderApp::~ReaderApp()
+{
+    RTPSDomain::removeRTPSParticipant(rtps_participant_);
+    delete(reader_history_);
+}
+
+bool ReaderApp::register_entity(std::string topic_name)
+{
+    std::cout << "Registering RTPS Reader" << std::endl;
+
+    TopicAttributes topic_att;
+    topic_att.topicKind = NO_KEY;
+    topic_att.topicDataType = "HelloWorld";
+    topic_att.topicName = topic_name;
+
+    eprosima::fastdds::dds::ReaderQos reader_qos;
+    reader_qos.m_durability.kind = eprosima::fastdds::dds::TRANSIENT_LOCAL_DURABILITY_QOS;
+    reader_qos.m_reliability.kind = eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS;
+
+    return rtps_participant_->registerReader(rtps_reader_, topic_att, reader_qos);
+}
```

Comment:
We may avoid this method if we introduce this code in the constructor.
What do you think? Consider apply this change in the writer too

Reply:
Yes, I introduced it to highlight the steps for the registration but I can add a comment

Reply:
Sure! Comments are very welcome!